portfolio/src/main/java/com/google/sps/servlets/PokedexDataServlet.java

```diff
@@ -0,0 +1,49 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/pokedex-data")
+public class PokedexDataServlet extends HttpServlet {
+  private static final int NUMBER_OF_GENERATIONS =
```

Comment:
Optional comment for later: You could automatically determine the number of generations from the data. E.g. figure out the max value for the "Generation" column.

That would make this code more resilient to changes in the data, i.e. if a new generation is introduced, it won't break. However, you can add this at a later point, if you want.